src/main/java/module/lineup/assistant/LineupAssistant.java

```diff
@@ -53,159 +53,250 @@ public final boolean isPlayerInStartingEleven(int spielerId, Vector<MatchLineupP
+			optimizeLineup(b, players, substitutePositions);
 		}
 
-		// fill remaining seats
-		for ( byte b : reservePositionOrder) {
-			doReserveSpielerAufstellen(b, bForm, bInjured, bSuspended, lPlayers, lPositions);
+		var backupPositionOrder = List.of(
+				List.of(IMatchRoleID.KEEPER),
+				List.of(IMatchRoleID.CENTRAL_DEFENDER),
+				List.of(IMatchRoleID.BACK),
+				List.of(IMatchRoleID.MIDFIELDER),
+				List.of(IMatchRoleID.WINGER),
+				List.of(IMatchRoleID.FORWARD),
+				List.of(IMatchRoleID.EXTRA)
+		);
+		var backupCandidates = new ArrayList<>(substitutePositions.stream()
+				.map(MatchLineupPosition::getPlayer)
+				.filter(Objects::nonNull)
+				.toList());
+		lPositions = allPositions.stream().filter(MatchRoleID::isBackupsMatchRoleId).toList();
+		for (var b : backupPositionOrder) {
+			// local backup candidates list without player of corresponding substitutes slot
+			var substitutePosition = getSubstitutePosition(b.get(0));
+			var matchLineupPosition = allPositions.stream().filter(i->i.getRoleId()==substitutePosition).findFirst();
+			if (matchLineupPosition.isPresent()) {
+				var substitutePlayerId = matchLineupPosition.get().getPlayerId();
+				var backupCandidatesForPosition = new ArrayList<>(backupCandidates.stream().filter(i->i.getPlayerId() != substitutePlayerId).toList());
+				optimizeLineup(b, backupCandidatesForPosition, lPositions);
+			}
 		}
 	}
 
-	private void addForward(ArrayList<Byte> fieldPlayerPositionOrder, ArrayList<Byte> reservePositionOrder) {
-		// FORWARD
-		fieldPlayerPositionOrder.add(IMatchRoleID.FORWARD);
-		fieldPlayerPositionOrder.add(IMatchRoleID.FORWARD_DEF);
-		fieldPlayerPositionOrder.add(IMatchRoleID.FORWARD_TOWING);
+	private int getSubstitutePosition(byte position) {
+		return switch (position){
+			case IMatchRoleID.KEEPER -> IMatchRoleID.substGK1;
+			case IMatchRoleID.CENTRAL_DEFENDER -> IMatchRoleID.substCD1;
+			case IMatchRoleID.BACK -> IMatchRoleID.substWB1;
+			case IMatchRoleID.MIDFIELDER -> IMatchRoleID.substIM1;
+			case IMatchRoleID.WINGER -> IMatchRoleID.substWI1;
+			case IMatchRoleID.FORWARD -> IMatchRoleID.substFW1;
+			case IMatchRoleID.EXTRA -> IMatchRoleID.substXT1;
+            default -> throw new IllegalStateException("Unexpected value: " + position);
+        };
+	}
 
-		reservePositionOrder.add(IMatchRoleID.FORWARD);
+	private void optimizeLineup(List<Byte> requestedPositions, ArrayList<Player> players, List<MatchLineupPosition> lineupPositions) {
+		if ((lineupPositions == null) || (players == null)) return;
+		for (var pos : lineupPositions) {
+			if (pos.getPlayerId() == 0 &&   // there isn't already a player at this position
+					requestedPositions.contains(pos.getPosition())) {
+
+				var behaviours = new ArrayList<>(MatchLineupPosition.getBehaviours(pos.getRoleId()));
+				if (behaviours.isEmpty()) {behaviours.add(IMatchRoleID.NORMAL);}
+				var maxRating = -1.0;
+				Player bestPlayer = null;
+				byte bestBehaviour = 0;
+				for (var behaviour : behaviours) {
```

Comment:
Wow!!  So this computes the best attitude for the player in the position?  That's pretty cool!

Reply:
Attitude or Behaviour? What would be the better name for it?

Reply:
> Attitude or Behaviour? What would be the better name for it?

Hattrick calls them “individual orders.” Behaviour is fine!